presentation/src/main/java/com/android254/presentation/speakers/SpeakerDetailsScreenViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package com.android254.presentation.speakers
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android254.domain.models.ResourceResult
+import com.android254.domain.repos.SpeakersRepo
+import com.android254.presentation.models.SpeakerUI
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+
+sealed interface SpeakerDetailsScreenUiState {
+
+    object Loading : SpeakerDetailsScreenUiState
+
+    data class Success(val speaker: SpeakerUI ) : SpeakerDetailsScreenUiState
+
+    data class Error(val message: String ) : SpeakerDetailsScreenUiState
+
+    data class SpeakerNotFound(val message: String):SpeakerDetailsScreenUiState
+}
+
+@HiltViewModel
+class SpeakerDetailsScreenViewModel @Inject constructor(
+    private val speakersRepo: SpeakersRepo,
+    private val savedStateHandle: SavedStateHandle,
+) :ViewModel(){
+
+    private val _uiState = MutableStateFlow<SpeakerDetailsScreenUiState>(SpeakerDetailsScreenUiState.Loading)
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        savedStateHandle.get<Int>("speakerId")?.let{
+            viewModelScope.launch {
+                getSpeakerById(id = it)
+            }
+        }
+    }
```

Comment:
let's avoid making calls in the init block

Reply:
Noted